compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -482,7 +479,10 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     }
 
     fn opt_local_def_id(&self, node: NodeId) -> Option<LocalDefId> {
```

Comment:
Needs a comment:

```suggestion
    /// Given the id of some node in the AST, finds the `LocalDefId` associated with it by the name resolver (if any),
    /// after applying any remapping from `get_remapped_def_id`.
    fn opt_local_def_id(&self, node: NodeId) -> Option<LocalDefId> {
```

...but really, it needs a better one. Does this correspond to the "def-id of the thing being defined at this node-id"? The def-id of the thing being referenced? Either one, depending on the kind of node-id?

I think the answer is the latter, as we tend to pun like that, but I'd have to go look at the calls and code to be sure.

Ideally we would probably document this on the `node_id_to_def_id` map itself and then refer to that field from here.

Reply:
`node_id_to_def_id` shouldn't be changed, just maps from node id to def id as usual. The thing here is that we would like to get from the "original" node id to the "final" def id. By original I meant, the node id of `'a` in your example and by final def id, I meant, the def id of `'a1` and not `'a`. So this will get node_id of `'a` and get it's def_id and then map from that def_id into `'a1` def_id.

Reply:
Fixed in https://github.com/rust-lang/rust/pull/99867/commits/d4d3dae04ccdc0d7a49846da7102e093ef6feeed